src/Microsoft.ML/Runtime/Internal/Tools/GeneratorUtils.cs

```diff
@@ -0,0 +1,467 @@
+﻿using Microsoft.CSharp;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.EntryPoints;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Newtonsoft.Json.Linq;
+using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.ML.Runtime.Internal.Tools
+{
+    internal static class GeneratorUtils
+    {
+        public static string GetFullMethodName(ModuleCatalog.EntryPointInfo entryPointInfo)
+        {
+            return entryPointInfo.Name;
+        }
+
+        public class EntryPointGenerationMetadata
+        {
+            public string Namespace { get; private set; }
+            public string ClassName { get; private set; }
+            public EntryPointGenerationMetadata(string @namespace, string className)
```

Comment:
As far as I see these are assigned only in the constructor. So perhaps we can just drop the `private set`.